tolstoy/src/tx_client.rs

```diff
@@ -62,52 +75,61 @@ pub struct TxClient {}
 
 impl TxReader for TxClient {
     fn all(sqlite: &rusqlite::Connection) -> Result<Vec<Tx>> {
-        // Make sure a=txInstant rows are first, so that we process
-        // all transactions before we process any transaction parts.
         let mut stmt = sqlite.prepare(
             "SELECT
                 e, a, v, tx, added, value_type_tag,
                 CASE a WHEN :txInstant THEN 1 ELSE 0 END is_transaction
```

Comment:
Strictly speaking this is incorrect: you would need to check:

* `a` is `:db/txInstant`
* `tx == e`

But, as noted below, I think it's better to do this in code.